sphinx/builders/linkcheck.py

```diff
@@ -301,10 +316,15 @@ def write_doc(self, docname: str, doctree: Node) -> None:
         if self.broken:
             self.app.statuscode = 1
 
-    def write_entry(self, what: str, docname: str, line: int, uri: str) -> None:
-        with open(path.join(self.outdir, 'output.txt'), 'a', encoding='utf-8') as output:
-            output.write("%s:%s: [%s] %s\n" % (self.env.doc2path(docname, None),
-                                               line, what, uri))
+    def write_entry(self, what: str, docname: str, docpath: str, line: int,
+                    uri: str) -> None:
```

Comment:
Why changing format of output.txt? Is it needed for this change?

Reply:
Because it is only necessary to calculate docpath/filename once; in process_result.

Reply:
So that it's not necessary to call `doc2path()` twice.

Reply:
Okay, I understand. If so, please replace `docname` parameter by new `filename` parameter. It seems `docname` is no longer used. So it is better.

Reply:
I've removed docname from the object in favor of just `filename`

Reply:
But not from the call to `write_entry` because people could be / could've been parsing that file?